From 17fb11291217df11dc47a41bcb8a3403fb1704ad Mon Sep 17 00:00:00 2001
From: Uri Lublin <uril@redhat.com>
Date: Wed, 12 Dec 2012 16:30:47 +0000
Subject: [PATCH] qxl+vnc: register a vm state change handler for dummy
 spice_server

When qxl + vnc are used, a dummy spice_server is initialized.
The spice_server has to be told when the VM runstate changes,
which is what this patch does.

Without it, from qxl_send_events(), the following error message is shown:
  qxl_send_events: spice-server bug: guest stopped, ignoring

Cc: qemu-stable@nongnu.org
Signed-off-by: Uri Lublin <uril@redhat.com>
Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
---
 ui/spice-core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ui/spice-core.c b/ui/spice-core.c
index ba0d0bd..cb4b8b9 100644
--- a/ui/spice-core.c
+++ b/ui/spice-core.c
@@ -775,6 +775,8 @@ int qemu_spice_add_interface(SpiceBaseInstance *sin)
          */
         spice_server = spice_server_new();
         spice_server_init(spice_server, &core_interface);
+        qemu_add_vm_change_state_handler(vm_change_state_handler,
+                                         &spice_server);
     }
 
     return spice_server_add_interface(spice_server, sin);
